### ISSUE TITLE
SQL editor highlights Redshift code incorrectly when apostrophes are correctly escaped

### ISSUE THREAD

**Issue by hlascelles** [dbeaver/dbeaver#10348]
## Issue

When an apostrophe value is escaped (correctly) the editor thinks the string or regex has not finished and highlights subsequent SQL commands as part of the string. In this case it is Postgres, but it might happen in other DB SQL windows.

**DBeaver version: 7.2.4.202011011603**

eg:

![image](https://user-images.githubusercontent.com/299102/99060470-fb386a80-2597-11eb-8489-725b9afbb255.png)

**Comment by ShadelessFox** [comment]
Hello, @hlascelles, please note that PostgreSQL uses single quote as an escape character. Instead of `\'` you should write `''`.
Let me know if the issue was solved.

**Comment by hlascelles** [comment]
My apologies, I see now it was a Redshift tab. I have updated the title!

I have reconfirmed that the above is a valid SQL command that runs correctly against Redshift, but is not highlighted correctly in DBeaver.

You are right though, changing the `\'` to a `''` still returns the right results. I am unsure now how to proceed with a fix for this. DBeaver is not highlighting correctly, it is true, but there is a "better way of doing the query".

It seems [some others](https://stackoverflow.com/a/54715905/1267203) report similar issues in other IDEs. 

I can conduct further tests if you like.
